File: 프로그래머스/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    baguni = []
    len_board = len(board)
    for i in board:
        print(i)
    for i in moves:
        print(baguni)
        for j in range(len_board):
            if board[j][i-1] != 0:
                baguni.append(board[j][i-1])
                board[j][i - 1] = 0
                break

        if len(baguni) >= 2:
            if baguni[-1] == baguni[-2]:
                baguni.pop(-1)
                baguni.pop(-1)
                answer += 2

    return answer


board = [[0, 0, 0, 0, 0], [0, 0, 1, 0, 3], [0, 2, 5, 0, 1], [4, 2, 4, 4, 2], [3, 5, 1, 3, 1]]
moves = [1, 5, 3, 5, 1, 2, 1, 4]

print(solution(board, moves))

File: JavaScript/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    var n = board.length
    // board.forEach((e)=>{
    //     console.log(e)
    // })
    // console.log()
    var new_board = []
    for (var j = 0; j < n; j++) {
        var tmp_line = []
        for (var i = n - 1; i >= 0; i--) {
            tmp_line.push(board[i][j])
        }
        new_board.push(tmp_line)
    }
    // new_board.forEach((e)=>{
    //     console.log(e)
    // })

    var box = []
    moves.forEach((e) => {
        while (new_board[e - 1].length > 0) {
            var doll = new_board[e - 1].pop()
            if (doll != 0) {
                if (box.length > 0 & box[box.length - 1] == doll) {
                    box.pop()
                    answer++

                } else {
                    box.push(doll)
                }
                break
            }
        }
        // console.log(box)
        // console.log(answer)
    })


    // new_board.forEach((e)=>{
    //     console.log(e)
    // })
    return answer * 2;
}